conjure-java-core/src/main/java/com/palantir/conjure/java/services/UndertowServiceHandlerGenerator.java

```diff
@@ -510,6 +507,78 @@ private CodeBlock decodePlainParameterCodeBlock(Type type, TypeMapper typeMapper
         );
     }
 
+    private Optional<CodeBlock> getComplexTypeDeserializer(
+            Type type, TypeMapper typeMapper, String resultVarName, String paramsVarName, String paramId) {
+        return type.accept(new TypeVisitor.Default<Optional<String>>() {
+            @Override
+            public Optional<String> visitExternal(ExternalReference value) {
+                return Optional.of("deserializeComplex");
+            }
+
+            @Override
+            public Optional<String> visitReference(com.palantir.conjure.spec.TypeName value) {
+                return Optional.of("deserializeComplex");
+            }
+
+            @Override
+            public Optional<String> visitPrimitive(PrimitiveType value) {
+                return Optional.empty();
+            }
+
+            @Override
+            public Optional<String> visitList(ListType value) {
+                return value.getItemType().accept(this).map(ignored -> "deserializeComplexList");
+            }
+
+            @Override
+            public Optional<String> visitSet(SetType value) {
+                return value.getItemType().accept(this).map(ignored -> "deserializeComplexSet");
+            }
+
+            @Override
+            public Optional<String> visitOptional(OptionalType value) {
+                return value.getItemType().accept(this).map(ignored -> "deserializeOptionalComplex");
+            }
+
+            @Override
+            public Optional<String> visitDefault() {
+                return Optional.empty();
+            }
+        }).map(functionName -> CodeBlocks.statement(
+                "$1T $2N = $3T.$4L($5N.get($6S), $7T::valueOf)",
+                typeMapper.getClassName(type),
+                resultVarName,
+                StringDeserializers.class,
+                functionName,
+                paramsVarName,
+                paramId,
+                typeMapper.getClassName(getComplexType(type))));
+    }
+
+    private Type getComplexType(Type type) {
```

Comment:
In conjure documentation, we've been using the term 'container' types to refer to list, set, map and optional.

https://palantir.github.io/conjure/#/docs/concepts?id=container-types

Reply:
Updated